### PR TITLE
Cfy 2872 nova creat validation fix

### PR DESCRIPTION
**Pull request by adamlavie**

### CODE REVIEW

File: nova_plugin/tests/test_server.py [hunk 1]
```diff
@@ -128,3 +128,22 @@ def mock_get_server_by_context(_):
...
+    @mock.patch('nova_plugin.server.start')
+    @mock.patch('nova_plugin.server._handle_image_or_flavor')
+    @mock.patch('nova_plugin.server._fail_on_missing_required_parameters')
+    def test_nova_server_creation_param_integrity(self, *_):
+        class MyDict(dict):
+            pass
+
+        def mock_create_server(_, **kwargs):
+            key_args = MyDict(kwargs)
+            key_args.id = 'uid'
```
Comment (adamlavie) [inline nova_plugin/tests/test_server.py]: move to MyDict class

File: nova_plugin/tests/test_server.py [hunk 2]
```diff
@@ -128,3 +128,22 @@ def mock_get_server_by_context(_):
...
+            self.assertEqual(key_args['scheduler_hints'],
```
Comment (adamlavie) [inline nova_plugin/tests/test_server.py]: also assert key is in key_args

File: nova_plugin/tests/test_server.py [hunk 3]
```diff
@@ -128,3 +128,22 @@ def mock_get_server_by_context(_):
...
+    @mock.patch('nova_plugin.server.start')
+    @mock.patch('nova_plugin.server._handle_image_or_flavor')
+    @mock.patch('nova_plugin.server._fail_on_missing_required_parameters')
+    def test_nova_server_creation_param_integrity(self, *_):
+        class MyDict(dict):
+            id = 'uid'
+
+        def mock_create_server(*args, **kwargs):
+            key_args = MyDict(kwargs)
+            self.assertTrue('scheduler_hints' in key_args)
```
Comment (ran-z) [inline nova_plugin/tests/test_server.py]: use "self.assertIn", it produces better error messages

File: nova_plugin/server.py [hunk 4]
```diff
@@ -228,41 +226,22 @@ def rename(name):
...
-                "properties.nova.instance)".format(k))
-
-    for k in params:
-        if k in server:
-            params[k] = server[k]
-
-    if not params['meta']:
-        params['meta'] = dict({})
+    if 'meta' not in server:
+        server['meta'] = dict({})
```
Comment (ran-z) [inline nova_plugin/server.py]: just use dict() instead of dict({})
(yes i know its been there before as well :))

**Pull request merged**